cpp/src/gandiva/precompiled/decimal_sql.h

```diff
@@ -0,0 +1,34 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef DECIMAL_SQL_H
+#define DECIMAL_SQL_H
+
+#include <cstdint>
+#include <string>
+#include "gandiva/decimal_full.h"
+
+namespace gandiva {
+namespace decimalsql {
+
+Decimal128 Add(const Decimal128Full& x, const Decimal128Full& y, int32_t out_precision,
```

Comment:
Add a docstring? Is this function solely meant to test the algorithm implemented in LLVM IR elsewhere?

Reply:
added docstring. used in testing and the entry-point for the IR function (from add_large_decimal128_decimal128 in decimal_wrapper.cc)